lib/api/apiUtils/object/versioning.js

```diff
@@ -144,29 +185,48 @@ export function versioningPreprocessing(bucketName, bucketMD, objectKey,
             // object does not exist or is not versioned (before versioning)
             options.versionId = '';
             options.isNull = true;
-            options.deleteData = true;
+            options.dataToDelete = mstObjLocation;
             return callback(null, options);
         }
         // TODO check AWS behaviour
```

Comment:
I deleted it; in any case we can't check AWS behavior because AWS also does not allow PUT to specific versions afaik.